map_gen/maps/crash_site/commands.lua

```
@@ -146,6 +146,39 @@ local function abort(_, player)
     end
 end
 
+local function spy(args, player)
+    local player_name = player.name
+    local coin_count = player.get_item_count("coin")
+
+    -- Parse the values from the location string
+    -- {location = "[gps=-110,-17,redmew]"}
+    local location_string = args.location
+    local coords = {}
+    local coord_count = 0
+    for m in string.gmatch( location_string, "%-?%d+" ) do
+        table.insert(coords, tonumber(m))
+        coord_count = coord_count + 1
+    end
+
+    -- Do some checks then reveal the pinged map and remove 1000 coins
+    if coord_count ~= 2 then
+        game.players[player_name].print("[color=red]You need to add a valid location to send a spy fish. e.g /spy [gps=-110,-17,redmew][/color]")
+        return
+    elseif coin_count < 1000 then
+        game.players[player_name].print("[color=red]Training these spy fish ain't cheap! You need more coins![/color]")
+        return
+    else
+        local xpos=coords[1]
+        local ypos=coords[2]
+        local rad=16
+        game.player.force.chart(game.player.surface, {{xpos-rad, ypos-rad}, {xpos+rad, ypos+rad}})
+        game.print("[color=red]Jayefuu used the /spy command and spent 1000 coins to train a fish to spy on the enemy [gps=" .. xpos .. "," .. ypos .. ",redmew][/color]")
+        local inv = game.player.get_inventory(1)
```

I assume 1 is `defines.inventory.character_main` either add a comment, use `defines.inventory.character_main` directly or make a variable that uses  `defines.inventory.character_main` either locally or at the top of the file.

---

I believe I've resolved this now. I moved inv to the top and then always use inv. So if the coins are in a different inventory then they won't be found and it will fail early.

---

I think you are confusing this with a different comment. My point here was that when reading the code it’s not easy to know what `get_inventory(1)` does, as opposed to `get_inventory(defines.inventory.character_main)`.